### PR TITLE
[#4137] docs(spark-connector): correct timestamp mapping document for Spark connector and Gravitino

### PR DESCRIPTION
**Pull request by lsyulong**
### What changes were proposed in this pull request?
correct timestamp mapping document for Spark connector and Gravitino

### Why are the changes needed?
Fix: #4137 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
just document

### CODE REVIEW

**Comment by FANNG1** [comment]
@lsyulong , some suggestions:
1. could you modify the PR description?  you could refer other PRs
2. There is no need to add `TIMESTAMPNTZ` type to Gravitino, it's ok to just modify document.
3. could you format the table in the document to keep align,  there's existing tools like `idea` to do this.

**Comment by lsyulong** [comment]
**What changes were proposed in this pull request?**
Modified the mapping relationship between Gravitino and Spark data types in the Spark connector section of the document

**Why are the changes needed?**

Because the mapping relationship described in the previous document was incorrect

Fix: https://github.com/apache/gravitino/issues/4137

File: docs/spark-connector/spark-connector.md [hunk 1]
```diff
@@ -78,23 +78,23 @@ The command `SHOW CATALOGS` will only display the Spark default catalog, named s
...
+| `ShortType`        | `short`             | 0.5.0         |
+| `IntegerType`      | `integer`           | 0.5.0         |
+| `LongType`         | `long`              | 0.5.0         |
+| `FloatType`        | `float`             | 0.5.0         |
+| `DoubleType`       | `double`            | 0.5.0         |
+| `DecimalType`      | `decimal`           | 0.5.0         |
+| `StringType`       | `string`            | 0.5.0         |
+| `CharType`         | `char`              | 0.5.0         |
+| `VarcharType`      | `varchar`           | 0.5.0         |
+| `TimestampType`    | `timestamp`         | 0.5.0         |
```
Comment (FANNG1) [inline docs/spark-connector/spark-connector.md:94]: please correct to:
`TimestampType` -> `timestamp with time zone` 
`TimestampNTZType` -> `timestamp without time zone`

**Comment by lsyulong** [comment]
Okay, I have corrected the mapping relationship between Spark and Gravitino data types

**Comment by lsyulong** [comment]
Hello, I don't just need to copy the content above, just modify it in the document. I only changed TimestampNTZType ->timestamp now. I misunderstood again

**Comment by FANNG1** [comment]
> Hello, I don't just need to copy the content above, just modify it in the document. I only changed TimestampNTZType ->timestamp now. I misunderstood again

`Timestamp` type in Spark means timestamp type with time zone while `TimestampNTZType` in spark means timestamp without time zone, we should correct it like this.

**Comment by lsyulong** [comment]
OK

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@lsyulong , merged to main, thanks for your contribution and hope you enjoy the journey.

**Pull request merged**